### PR TITLE
Adapt beluga_tutorial code to key concepts

### PR DESCRIPTION
**Pull request by Alondruck**
### Proposed changes
Adapt the main code to the names used by the key concepts.

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [ ] 🚀 Feature (change which adds functionality)
- [x] 📚 Documentation (change which fixes or extends documentation)

### Checklist

- [x] Lint and unit tests (if any) pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### Additional comments
This PR will wait for #390 to be merged into the main branch.

### CODE REVIEW

**Comment by Alondruck** [comment]
@nahueespinosa @hidmic I realized that we are not using the default values of the Parameters struct. Should we use them, or is it fair to require the user to provide a path to the parameters file?

In the explanation of the tutorial, I'll delete all the code related to YAML, and the user will use the default parameters. So, I want to know your thoughts on whether we also need to let the user use the default parameters here.

File: beluga_tutorial/src/main.cpp [hunk 1]
```diff
@@ -224,15 +222,14 @@ int run(const std::filesystem::path& path) {
...
     particles |= beluga::actions::propagate(std::execution::seq, motion_model);
-    record.propagate = particles;
+    record.prediction = particles;
 
     particles |= beluga::actions::reweight(std::execution::seq, sensor_model) | beluga::actions::normalize;
-    record.reweight = particles;
 
     particles |= beluga::views::sample |                                        //
                  ranges::views::take_exactly(parameters.number_of_particles) |  //
                  beluga::actions::assign;
```
Comment (nahueespinosa) [inline beluga_tutorial/src/main.cpp:231]: nit: We can define the update stage like this:

```
particles |= beluga::actions::reweight(std::execution::seq, sensor_model) |  //
             beluga::actions::normalize |                                    //
             beluga::views::sample |                                         //
             ranges::views::take_exactly(parameters.number_of_particles) |   //
             beluga::actions::assign;
```

**Review by nahueespinosa — COMMENTED** [review]
> I want to know your thoughts on whether we also need to let the user use the default parameters here.

No strong opinion on this one. We could make the parameter_path argument optional, loading the default values if no yaml file is provided, but I also don't mind that we always ignore the default values.

File: beluga_tutorial/src/main.cpp (hunk 1 above)
Comment (Alondruck) [reply beluga_tutorial/src/main.cpp:231]: Maybe in this code, we can write the update stage this way. However, in the explanation of the code, I still want to separate the stages for explanatory purposes.

**Review by nahueespinosa — APPROVED** [review]
LGTM!

**Pull request merged**